### PR TITLE
Resize indexing misc bug fixes

### PR DESCRIPTION
**Pull request by naoyam**
Just minor bug fixes. Please see the inline comments.

### CODE REVIEW

**Comment by naoyam** [comment]
!test

File: csrc/id_model/indexing_traversal.cpp [hunk 1]
```diff
@@ -65,18 +86,26 @@ std::optional<IndexingTraversal::ExprPath> IndexingTraversal::
       /*build_graphs=*/false);
 
   // Gather all resize exprs for each of the inputs and outputs
-  std::unordered_map<Val*, std::vector<Resize*>> tv_resize_map;
-  for (auto inp : ir_utils::filterByType<TensorView>(expr->inputs())) {
-    for (auto expr : inp->domain()->allExprs()) {
+  std::unordered_map<TensorView*, std::vector<Resize*>> tv_resize_map;
+  for (auto inp : expr->inputs()) {
```
Comment (naoyam) [inline csrc/id_model/indexing_traversal.cpp:90]: This was a trivial bug that I assumed `expr->inputs()` are still `TensorView` and not `TensorIndex`, which is not true for predicating vectorized ops.

File: csrc/id_model/indexing_traversal.cpp [hunk 2]
```diff
@@ -44,6 +44,27 @@ IndexingTraversal::IndexingTraversal(
     }
     resize_paths_.insert(resize);
   }
+
+  // A unique expr path should be always allowed
```
Comment (naoyam) [inline csrc/id_model/indexing_traversal.cpp:48]: An war for a couple of issues I encountered while playing with RoPE patterns.

File: csrc/id_model/indexing_traversal.cpp [hunk 3]
```diff
@@ -44,6 +44,27 @@ IndexingTraversal::IndexingTraversal(
...
+  for (const auto& expr_g : graph.disjointExprSets().disjointSets()) {
+    auto resize = dynamic_cast<Resize*>(expr_g->front());
+    if (resize == nullptr) {
+      continue;
+    }
+
+    auto input_groups = graph.inputGroups(expr_g);
+    auto output_groups = graph.outputGroups(expr_g);
+    if (input_groups.size() != 1 || output_groups.size() != 1) {
```
Comment (jacobhinkle) [inline csrc/id_model/indexing_traversal.cpp:57]: At first I thought this should be an `NVF_ERROR` but I guess this handles the case where there a `Resize` output is mapped to an ID that was another Resize, or a Split/Merge from a reshape. Is that correct?

**Review by jacobhinkle — APPROVED** [review]
LGTM

File: csrc/id_model/indexing_traversal.cpp (hunk 3 above)
Comment (naoyam) [reply csrc/id_model/indexing_traversal.cpp:57]: I actually don't remember why I chose to do this. `NVF_ERROR` seems to make more sense. Let me check.

Comment (naoyam) [reply csrc/id_model/indexing_traversal.cpp]: Assertion seems fine. Changed.

**Comment by naoyam** [comment]
!build

**Pull request merged**